pingpong/ud_pingpong.c

```diff
@@ -449,16 +449,26 @@ static int server_connect(void)
 {
 	int ret;
 	struct fi_cq_entry comp;
+	struct timespec a, b;
 
 	ret = common_setup();
 	if (ret != 0)
 		goto err;
 
+	clock_gettime(CLOCK_REALTIME_COARSE, &a);
```

Comment:
How about adding an `if (timeout > 0)` guard here to keep the unconditional `clock_gettime` overhead from showing up in non-timeout runs?